File: 38-外观数列.cpp
class Solution {
public:
    string countAndSay(int n) {
        string cur_str = "1";
        string next_str = "";
        int j, num;
        
        for(int i = 1; i < n; i++){
            j = 0;
            while(j < cur_str.size()){
                num = 1;
                while(j < cur_str.size() - 1 && cur_str[j + 1] == cur_str[j]){
                    num++;
                    j++;
                }
                next_str.push_back((char)(num + '0'));
                next_str.push_back(cur_str[j]);
                j++;
            }
            cur_str = next_str;
            next_str = "";
        }

        return cur_str;
    }
};

File: 38-外观数列.py
class Solution:
    def countAndSay(self, n: int) -> str:
        cur_str = '1'
        next_str = ''

        for i in range(1, n):
            j = 0
            while j < len(cur_str):
                num = 1
                while j < len(cur_str) - 1 and cur_str[j + 1] == cur_str[j]:
                    j += 1
                    num += 1
                next_str += str(num) + cur_str[j]
                j += 1
            cur_str = next_str
            next_str = ''
        
        return cur_str